### PR TITLE
Remove redundant boolean state from LongDecimalWithOverflowAndLongState

### PR DESCRIPTION
**Pull request by raunaqmorarka**
## Description
Remove redundant boolean state from LongDecimalWithOverflowAndLongState

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/operator/aggregation/state/LongDecimalWithOverflowAndLongStateFactory.java [hunk 1]
```diff
@@ -66,27 +79,80 @@ public void addLong(long value)
...
-        protected long longValue;
+        private final long[] unscaledDecimal = new long[2];
+        private long longValue;
+        private long overflow;
 
         public SingleLongDecimalWithOverflowAndLongState() {}
 
         // for copying
-        private SingleLongDecimalWithOverflowAndLongState(long longValue)
+        private SingleLongDecimalWithOverflowAndLongState(long[] unscaledDecimal, long longValue, long overflow)
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/operator/aggregation/state/LongDecimalWithOverflowAndLongStateFactory.java:151]: I don't understand. Was this a bug? It wasn't delegating to super constructor.

Does this deserve a separate commit/PR?

**Review by sopel39 — APPROVED** [review]
lgtm % question about bug

File: core/trino-main/src/main/java/io/trino/operator/aggregation/state/LongDecimalWithOverflowAndLongStateFactory.java (hunk 1 above)
Comment (raunaqmorarka) [reply core/trino-main/src/main/java/io/trino/operator/aggregation/state/LongDecimalWithOverflowAndLongStateFactory.java:151]: It looked like a mistake in the current code to me but I don't know enough about how it is used to definitively say whether it would've resulted in a bug.
@kasiafi could you please take a look and clarify.

**Review by lukasz-stec — APPROVED** [review]
lgtm
